rust/wal3/src/reader.rs

```diff
@@ -223,17 +237,25 @@ impl LogReader {
             .filter(|f| ranges_overlap(log_position_range, (f.start, f.limit)))
             .cloned()
             .collect::<Vec<_>>();
-        Some(Self::post_process_fragments(fragments, from, limits))
+        let mut short_read = false;
+        Some(Self::post_process_fragments(
+            fragments,
+            from,
+            limits,
+            &mut short_read,
+        ))
     }
 
     fn post_process_fragments(
```

Comment:
Can we document what a `short_read` here means? it seems a bit nuanced so methinks that future us might wonder what this flag does. Also, boolean args always make me a little suspect -- is there a way to make the interfaces more explicit to remove the need to pass this flag? This flag is also mutable so does this mean the callers will expect this method to mutate it according to the logic inside here so that it can be used later?

Sorry if this is a dumb question. My Rust is very sub-par :/